src/compilers/vyper/output.rs

```diff
@@ -0,0 +1,159 @@
+use super::VyperCompilationError;
+use crate::artifacts::{BytecodeObject, Contract, Evm, FileToContractsMap, SourceFile};
+use alloy_json_abi::JsonAbi;
+use alloy_primitives::Bytes;
+use serde::Deserialize;
+use std::{
+    collections::{BTreeMap, HashSet},
+    path::{Path, PathBuf},
+};
+
+/// Before Vyper 0.4 source map was represented as a string, after 0.4 it is represented as a map
+/// where compressed source map is stored under `pc_pos_map_compressed` key.
+fn deserialize_sourcemap<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    #[derive(Deserialize)]
```

Comment:
should this be untagged?

Reply:
ah right, should've added tests right away

addressed in [2632f99](https://github.com/foundry-rs/compilers/pull/134/commits/2632f99f99789a7b3ef4ebf4126cd768daaee682)